clients/go/coreutils/literals.go

```diff
@@ -483,7 +484,17 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 		}
 	case *core.LiteralType_Simple:
 		newT := t.Type.(*core.LiteralType_Simple)
-		lv, err := MakeLiteralForSimpleType(newT.Simple, fmt.Sprintf("%v", v))
+		strValue := fmt.Sprintf("%v", v)
+		if newT.Simple == core.SimpleType_STRUCT {
+			if _, isValueStringType := v.(string); !isValueStringType {
```

Comment:
if v is a string, why do we try to json marshal it? won't that escape characters and cause havoc?

v can also be a map[string]interface{}, right? or just any type really... that we should json.Marshal... and json unmarshal into a pb struct.. right?

Reply:
Hi @EngHabu , we dont actually do json  marshal if its a string .  the condition is ! isValueStringType so we skip over if its a string  and pass it as it is to jsonpb.Unmarshal to unmarshal it to a structpb.

And yes v can be map[string]interface{} , which is what we are using from flytectl and that one will be json Marshalled and unmarshalled to pb struct.

Reply:
Check this test `Generic Passed As String` https://github.com/flyteorg/flyteidl/pull/176/files/39b9809db14ba0fd508958d72534580d4a453dc9#diff-87d3befa5f3d7a578349048f3e4711bcaee22606ce57a9339a8fd910e27e7168R136

And this one which passes map[string]interface 
https://github.com/flyteorg/flyteidl/pull/176/files/39b9809db14ba0fd508958d72534580d4a453dc9#diff-87d3befa5f3d7a578349048f3e4711bcaee22606ce57a9339a8fd910e27e7168R109

Any other type or invalid string will cause failure during unmarshal which we relying for throwing an error.

Reply:
👍🏾